codalab/server/rest_server.py

```diff
@@ -41,6 +43,77 @@
 ]
 
 
+class RequestCounter(object):
```

Comment:
I have no idea why you're doing this. If someone really wanted to DDOS us they'd get around your REMOTE_ADDR check.

Do you expect some existing code to send a ton of requests?